framework/src/modules/chain/state_store/account_store.js

```diff
@@ -51,7 +51,7 @@ class AccountStore {
 	async cache(filter) {
 		const result = await this.account.get(filter, { extended: true }, this.tx);
 		this.data = _.uniqBy([...this.data, ...result], this.primaryKey);
-		return _.cloneDeep(this.data);
+		return result;
```

Comment:
Since account state will be possible for mutation, it would be better to return the cloned version of the result

Reply:
AFAIK the data comes straight from the database and I see no additional reference anywhere, making a clone seems unnecessary. One could argue there is never too much safety but account will carry assets, possibly a lot, so we should consider the performance aspect.

Of course if you still believe it's necessary I'll make the change.
Note that [transaction store does not clone the result](https://github.com/LiskHQ/lisk-sdk/blob/development/framework/src/modules/chain/state_store/transaction_store.js#L35).

Reply:
If someone mutates the account in the `prepare` step, it will be mutated directly without using `set` function.
Meanwhile, whenever account is `get`, it is cloned.
Therefore, I think it should be consistent.

Reply:
Yeah, it needs to be cloned.

REF.: https://github.com/LiskHQ/lisk-sdk/issues/3913

Reply:
Okay I have read some code. This is going to be a discussion.

From what I understand you manage an internal initial copy so you can find out what has been modified. The only purpose I've found to this is so you can make a partial update later on.

Knowing that most modifications will be made to the asset field, which you won't be able to update partially since it's a userland structure.

Based on those element I don't see the purpose of all this mechanics. But I see you have to workaround this design with the extensive use of deep clone.

What am I missing ?

Reply:
the copying is not really related to how it's handled for the partial update structure.
The design that we provide is if you want to make any changes to account state from the custom transaction, it needs to use `set`.
However, if you pass back the reference, in other part of code, it can be changed unknowingly.
To avoid that, we want to ensure that we return by value

Reply:
Ok. I've updated the PR